hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+  }
+}
+{{< /wavejson >}}
+
+### Command Length and TXDATA/RXDATA Byte-masking (or Present Lack Thereof)
+
+There is no restriction on the length of a command.
+Even though the TXDATA and RXDATA registers require 4-byte accesses, any unused bytes in the TXDATA fifo (or unused slots in the RXDATA fifo), are simply ignored at the end of a command.
+There is no need to explicitly pad commands to use up extra TXDATA bytes.
+If the last few received bytes do not fill an entire RXDATA word, they will be posted to RXDATA at the end of the command and any unneeded bytes the end of the RXDATA word will be left as zeros.
+
+The following waveform illustrates an example SPI transaction, where neither the data transmitted nor the data received fit into an even number of 32-bit words.
+In this example, `A[31:0]` and `B[31:0]`, have been previously loaded into {{< regref "TXDATA" >}}, and afterwards one word, `X[31:0]` is available afterwards in {{< regref "RXDATA" >}}.
+
+{{< wavejson >}}
+{signal: [
+  ["ByteOrder=0",
+  {name: "sd[0]", wave: "x222222223344552233z.22x", data: ["A[31]", "A[30]", "A[29]", "A[28]", "A[27]","A[26]", "A[25]", "A[24]", "A[23]", "A[19]", "A[15]", "A[11]",
+                                                         "A[7]", "A[3]", "B[31]", "B[27]", "B[23]", "B[19]", "X[31]", "X[27]"]},
+  {name: "sd[1]", wave: "xz.......3344552233z.22x", data: ["A[22]", "A[18]", "A[14]", "A[10]", "A[6]", "A[2]", "B[30]", "B[26]", "B[22]", "B[18]", "X[30]", "X[26]"]},
+  {name: "sd[2]", wave: "xz.......3344552233z.22x", data: ["A[21]", "A[17]", "A[13]", "A[9]", "A[5]", "A[1]", "B[29]", "B[25]", "B[21]", "B[17]", "X[29]", "X[25]"]},
+  {name: "sd[3]", wave: "xz.......3344552233z.22x", data: ["A[20]", "A[16]", "A[12]", "A[8]", "A[4]", "A[0]", "B[28]", "B[24]", "B[20]", "B[16]", "X[28]", "X[24]"]},
+],
+   {name:""},
+   ["ByteOrder=1",
+  {name: "sd[0]", wave: "x555555554433225544z.55x", data: ["A[7]", "A[6]", "A[5]", "A[4]", "A[3]", "A[2]", "A[1]", "A[0]",
+                                                           "A[15]", "A[11]", "A[23]", "A[19]", "A[31]", "A[27]", "B[7]", "B[3]", "B[15]", "B[11]", "X[7]", "X[3]"]},
+  {name: "sd[1]", wave: "xz.......4433225544z.55x", data: ["A[14]", "A[10]", "A[22]", "A[18]", "A[30]", "A[26]", "B[6]", "B[2]", "B[14]", "B[10]", "X[6]","X[2]"]},
+  {name: "sd[2]", wave: "xz.......4433225544z.55x", data: ["A[13]", "A[9]", "A[21]", "A[17]", "A[29]", "A[25]", "B[5]", "B[1]", "B[13]", "B[9]", "X[5]", "X[1]"]},
+  {name: "sd[3]", wave: "xz.......4433225544z.55x", data: ["A[12]", "A[8]", "A[20]", "A[16]", "A[28]", "A[24]", "B[4]", "B[0]", "B[12]", "B[8]", "X[4]", "X[0]"]},
+  ],
+  ],
+  head: {
+    text: "Serial bit ordering for 6 bytes transmitted from FIFO words 'A[31:0]' and 'B[31:0]', and 1 byte received into word 'X[31:0]'",
+  },
+  foot: {
+    text: "Note: SPI_HOST command sent with COMMAND.TX1_CNT = 1, .TX4_CNT = 5, .DUMMY_CNT = 2, and  .RX_CNT=1"
```

Comment:
i'm really confused by this diagram. I don't see `TX4_CNT` defined anymore.
But can I assume it means the number of bytes to burst in quad mode?

Reply:
Hmm.  That should be TXN_CNT.  Fixed. Sorry about that.